org.eclipse.draw2d/src/org/eclipse/draw2d/geometry/Interval.java

```diff
@@ -0,0 +1,293 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2023 Google, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Google, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.draw2d.geometry;
+
+import java.io.Serializable;
+
+/**
+ * Represents interval in 1-dimensional space.
+ *
+ * @author mitin_aa
+ * @author lobas_av
+ * @author scheglov_ke
+ * @coverage gef.draw2d
+ */
+public class Interval implements Serializable {
+	private static final long serialVersionUID = 0L;
+	public static final Interval INFINITE = new Interval(0, Integer.MAX_VALUE);
+	public int begin;
+	public int length;
+
+	/**
+	 * Creates a new Interval with zero length and starting at the origin.
+	 * 
+	 * @since 3.13
+	 */
+	public Interval() {
+	}
+
+	/**
+	 * Creates a new Interval with the given {@code begin} and {@code length}
+	 * 
+	 * @param begin  The begin of this interval.
+	 * @param length The length of this interval.
+	 * @since 3.13
+	 */
+	public Interval(int begin, int length) {
+		this.begin = begin;
+		this.length = length;
+	}
+
+	/**
+	 * Creates a new Interval exact same parameters as the given Interval.
+	 * 
+	 * @param interval The interval to copy
+	 * @since 3.13
+	 */
+	public Interval(Interval interval) {
+		begin = interval.begin;
+		length = interval.length;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (o instanceof Interval) {
```

Comment:
if you prefer to not have Interval not final I think comparing the classes would be better. 

Should we add an this == o check?

Reply:
The equals method is now generated directly by Java.